circle.yml

```diff
@@ -671,6 +679,87 @@ commands:
+          command: |
+            cd render-test/ios && ./setup_test_data.sh
+            cmake ../../next -B Build -G Xcode -DCMAKE_SYSTEM_NAME=iOS -DCMAKE_OSX_SYSROOT=iphoneos -DIOS_RENDER_TESTING=ON
+            cd Build
+            xcodebuild -sdk iphoneos -project Mapbox\ GL\ Native.xcodeproj -scheme RenderTestApp build-for-testing -arch arm64 CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO -configuration Release
+            xcodebuild -sdk iphoneos -project Mapbox\ GL\ Native.xcodeproj build -target RenderTestAppTests -arch arm64 CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO -configuration Release
+      - run:
+          name: Codesign RenderTestApp for running on a physical device
+          command: |
+            cd render-test/ios
+            ./codesigning/generate-entitlements.swift
+            ls codesigning
+            cd Build
+            ../codesigning/codesign_all.sh
+            codesign -dv --verbose=4 Release-iphoneos/RenderTestApp.app # for debugging that app is signed or not
+      - run:
+          name: Login to IOS Google Cloud Platform
+          command: |
+            if [[ -n "${GCLOUD_SERVICE_ACCOUNT_JSON_IOS}" ]]; then
+              echo "${GCLOUD_SERVICE_ACCOUNT_JSON_IOS}" > iosAccount.json
+              gcloud auth activate-service-account --key-file iosAccount.json --project ios-mapbox-gl-native
+              rm iosAccount.json
+            fi
+      - run:
+          name: Run IOSRenderTestApp Tests on Firebase
+          no_output_timeout: 20m
+          command: |
+            if [[ -n "${GCLOUD_SERVICE_ACCOUNT_JSON_IOS}" ]]; then
+              # arrange files in a way Firebase expects it, and package them in a zip file
+              cd render-test/ios/Build
+              cp ../RenderTestApp_iphoneos13.2-arm64e-release.xctestrun .
+              zip testapp.zip -r Release-iphoneos/RenderTestApp.app
+              zip testapp.zip RenderTestApp_iphoneos13.2-arm64e-release.xctestrun
+              gcloud firebase test ios run \
+                --test testapp.zip \
+                --device model=iphonexs,version=12.1,locale=en,orientation=portrait --xcode-version=11.1 --timeout 20m \
+                --verbosity=debug --no-record-video --results-dir ios-render-test-app-${CIRCLE_BUILD_NUM}
+            fi
+      - run:
+          name: Retrieve Test Result from gcloud
+          when: always
+          command: |
+            if [[ -n "${GCLOUD_SERVICE_ACCOUNT_JSON_IOS}" ]]; then
+              testResult=$(gsutil ls -d gs://test-lab-hn7kwqcz3108z-hjm4qtbqqkymk/ios-render-test-app-${CIRCLE_BUILD_NUM}/iphonexs-12.1-en-portrait/TestLogs/)
+              mkdir -p render-test-app
+              gsutil -m cp -r $testResult render-test-app
+              cd render-test-app/TestLogs && ls
+              xcparse attachments *.xcresult
+              cd ../.. && mkdir -p /tmp/tests/ios-render-test
+              cp render-test-app/TestLogs/*.html /tmp/tests/ios-render-test  
```

Comment:
Metrics would be added in follow-up prs?

Reply:
yes!